plugin/src/main/scala/chisel3/internal/plugin/BundleComponent.scala

```diff
@@ -177,21 +177,34 @@ private[plugin] class BundleComponent(val global: Global, arguments: ChiselPlugi
       elementsImpl
     }
 
-    def generateAutoTypename(bundle: ClassDef, thiz: global.This, conArgsOpt: Option[List[Tree]]): Option[Tree] = {
-      conArgsOpt.map { conArgs =>
-        // Create an iterable out of all constructor argument accessors
-        val typeNameConParamsSym =
-          bundle.symbol.newMethod(
-            TermName("_typeNameConParams"),
-            bundle.symbol.pos.focus,
-            Flag.OVERRIDE | Flag.PROTECTED
+    def generateAutoTypename(record: ClassDef, thiz: global.This, conArgsOpt: Option[List[Tree]]): Option[Tree] = {
+      conArgsOpt.flatMap { conArgs =>
+        val recordIsAnon = record.symbol.isAnonOrRefinementClass
+
+        // Anonymous `Records` in any scope are forbidden
+        if (record.symbol.isAnonOrRefinementClass) {
+          global.reporter.error(
+            record.pos,
+            "Users can not mix 'HasAutoTypename' into an anonymous Record. Pull the Record definition into its own outer class."
```

Comment:
```suggestion
            "Users cannot mix 'HasAutoTypename' into an anonymous Record. Create a named class."
```
Previous error was conflating anonymity with inner-outer, we just need them to give the Record a name.